analyzer/codechecker_analyzer/analyzers/clangtidy/analyzer.py

```diff
@@ -203,6 +179,28 @@ def get_checker_list(self, config) -> Tuple[List[str], List[str]]:
             elif state == CheckerState.disabled:
                 checkers.append('-' + checker_name)
 
+        # -checks=-clang-analyzer-* option is added to the analyzer command by
+        # default except when all analyzer config options come from .clang-tidy
+        # file. The content of this file overrides every other custom config.
+        if config.analyzer_config.get('take-config-from-directory') == 'true':
```

Comment:
We should initiate a discussion about the necessity of this `take-config-from-directory` option. It was introduced because the default value of `--analyzer-config` flag is `['clang-HeaderFilterRegex=.*']`. This option enables clang-tidy reports in header files too. However, this option should be added where clang-tidy options are handled under `analyzers` directory. Merging clang-tidy configs from command line, config file, etc. should be simple, because this is just yaml data.